modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/60_wifi.js

```diff
@@ -326,6 +329,24 @@ return baseclass.extend({
 					])
 				];
 
+				if (bss.vlan) {
+					var desc = bss.vlan.getI18n();
+					var vlan_network = bss.vlan.getNetwork();
+					var vlan_zone;
+
+					if (vlan_network)
+						for (let zone of zones)
+							if (zone.getNetworks().includes(vlan_network))
+								vlan_zone = zone;
+
+					row[0].insertBefore(
+						E('div', {
+							'class' : 'zonebadge',
+							'title' : desc,
+							'style' : firewall.getZoneColorStyle(vlan_zone)
+						}, [ desc ]), row[0].firstChild);
```

Comment:
What happens if the code ends up at `firewall.getZoneColorStyle(vlan_zone)` and `vlan_zone` is still null? This code allows for that and it looks like `getZoneColorStyle` would crash with a null pointer in that case.

Reply:
If getZoneColorStyle cannot find the zone it returns a style with #eeeeee (light gray), which is the intended behavior (same as in the interface page if there is no firewall assigned to an interface)

```
	getZoneColorStyle: function(zone) {
		var hex = (zone instanceof Zone) ? zone.getColor() : getColorForName((zone != null && zone != '*') ? zone : null);
...
```

function getColorForName(forName) {
	if (forName == null)
		return '#eeeeee';
...
```

Reply:
Nice, I misread the code and placed the `null` outside the nested parenthesis, then it's fine.